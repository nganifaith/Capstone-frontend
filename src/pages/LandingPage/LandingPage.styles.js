import styled from 'styled-components';
import image from '../../images/background.jpg';

export const Wrapper = styled.div`
  width: 100vw;
  justify-content: center;

  a {
    background-color: #ff5617;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    border-radius: 50px;
    transition-duration: 0.4s;
    margin: 5px;

    :hover {
      box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24),
        0 17px 50px 0 rgba(0, 0, 0, 0.19);
    }
  }
`;

export const Header = styled.div`
  background-image: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0.25),
      rgba(0, 0, 0, 0.5)
    ),
    url(${image});
  height: 500px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  margin-top: -70px;
  padding-top: 70px;
`;

export const HeaderContent = styled.div`
  max-width: var(--max-width);
  margin: auto;
  text-align: left;
  color: var(--main-bg);
  padding: 70px var(--default-x-padding);

  h1 {
    max-width: 100%;
    width: 250px;
  }

  h4 {
    text-transform: capitalize;
    color: white;
    justify-content: start;
    font-weight: normal;
    max-width: 100%;
    width: 400px;
  }
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  color: black;
  
  & > div:first-child {
    max-width: var(--max-width);
    padding: 70px var(--default-x-padding) 0;
    margin: auto;
  }

  h2 {
    text-transform: uppercase;
    opacity: 0.5;
    font-size: 12px;
  }
`;

export const StatsContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto;
  max-width: var(--max-width);
  padding: 0 var(--default-x-padding) 70px;
  flex-wrap: wrap;

  & > * {
    flex: 1;
    margin: 20px;
    text-align: left;
  }
`;

export const Image = styled.img`
  max-width: 550px;
  width: 100%;
  margin-bottom: 10px;
`;
export const VisitContent = styled.div`
  background: #e7f0ff;

  & > div {
    max-width: var(--max-width);
    padding: 70px var(--default-x-padding);
    margin: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;

    & > * {
      flex: 1;
      margin: 20px;
      text-align: right;
    }
  }
`;

export const UserSection = styled.div`
  display: flex;
  flex-wrap: nowrap;
  background: #f8f7f7;
  overflow: scroll;
  padding: 70px;
`;

export const LastSection = styled.div`
  background-image: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0.35),
      rgba(0, 0, 0, 0.25)
    ),
    url(${image});
  align-content: center;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  padding: 70px;
  color: var(--main-bg);
`;
