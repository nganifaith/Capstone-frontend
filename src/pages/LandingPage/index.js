import React from 'react';
import { Link } from 'react-router-dom';
import img from '../../images/stats.png';
import nature from '../../images/nature.png';
import {
  Content,
  Header,
  HeaderContent,
  Image,
  StatsContent,
  VisitContent,
  Wrapper,
} from './LandingPage.styles';

const LandingPage = () => (
  <Wrapper>
    <Header>
      <HeaderContent>
        <h1> Reconnect With Nature</h1>
        <h4>
          There’s a whole world out there, right outside your window. You’d be a
          fool to miss it.
        </h4>
        <Link to="/login">Start for free</Link>
      </HeaderContent>
    </Header>
    <Content>
      <div>
        <h2>Best way to spend you Vacation</h2>
        <h1>Comfortable Sites to visit</h1>
      </div>

      <StatsContent>
        <div>
          <h1>All Information you need </h1>
          <p>
            Our yearly visits stats. Focusing a `background-image` on a Precise
            Location with Percentages
          </p>
          <Link to="/login">Start for free</Link>
        </div>
        <Image src={img} />
      </StatsContent>
      <VisitContent>
        <Image src={nature} />
        <div>
          <h1>Relax with a beautiful view </h1>
          <p>
            Our yearly visits stats. Focusing a `background-image` on a Precise
            Location with Percentages
          </p>
          <Link to="/login">Start for free</Link>
        </div>
      </VisitContent>
    </Content>
  </Wrapper>
);

export default LandingPage;
