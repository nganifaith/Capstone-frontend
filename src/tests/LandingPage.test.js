import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createStore } from 'redux';
import rootReducer from '../reducers';

import LandingPage from '../pages/LandingPage';

const history = createMemoryHistory();
const store = createStore(rootReducer);

test('full app rendering/navigating', () => {
  const welcome = renderer.create(
    <Provider store={store}>
      <Router history={history}>
        <LandingPage />
      </Router>
      ,
    </Provider>,
  ).toJSON();
  expect(welcome).toMatchSnapshot();
});
