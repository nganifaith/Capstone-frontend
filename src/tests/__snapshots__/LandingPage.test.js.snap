// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full app rendering/navigating 1`] = `
Array [
  <div
    className="sc-bdfBQB cqZURb"
  >
    <div
      className="sc-gsTEea fCeLxr"
    >
      <div
        className="sc-dlfnuX cVdvrT"
      >
        <h1>
           Reconnect With Nature
        </h1>
        <h4>
          There’s a whole world out there, right outside your window. You’d be a fool to miss it.
        </h4>
        <a
          href="/login"
          onClick={[Function]}
        >
          Start for free
        </a>
      </div>
    </div>
    <div
      className="sc-hKgJUU lhtyMn"
    >
      <div>
        <h2>
          Best way to spend you Vacation
        </h2>
        <h1>
          Comfortable Sites to visit
        </h1>
      </div>
      <div
        className="sc-eCstlR EFhEE"
      >
        <div>
          <h1>
            All Information you need 
          </h1>
          <p>
            Our yearly visits stats. Focusing a \`background-image\` on a Precise Location with Percentages
          </p>
          <a
            href="/login"
            onClick={[Function]}
          >
            Start for free
          </a>
        </div>
        <img
          className="sc-jSgvzq eqmZwv"
          src="stats.png"
        />
      </div>
      <div
        className="sc-gKseQn gvslRo"
      >
        <div>
          <img
            className="sc-jSgvzq eqmZwv"
            src="nature.png"
          />
          <div>
            <h1>
              Relax with a beautiful view 
            </h1>
            <p>
              Our yearly visits stats. Focusing a \`background-image\` on a Precise Location with Percentages
            </p>
            <a
              href="/login"
              onClick={[Function]}
            >
              Start for free
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-iBPTik dJJYOz"
    />
    <div
      className="sc-fubCzh hquMjL"
    >
      <h3>
        Connect with Nature
      </h3>
      <a
        href="/login"
        onClick={[Function]}
      >
        Start for free
      </a>
    </div>
  </div>,
  ",",
]
`;
